feat:增加百度百聘的抓取

sunshinenny opened the pull request:
## 制作该路由的想法
- 自己也大三了。需要为未来做些考虑
- 希望自己写的抓取可以帮到广大有需要的朋友们

## 路由存在的问题
- 希望可以加上每隔一天抓取，可惜找不到参考文档。无法实现
- 反爬严格的不行，可能访问的时候返回403，超时等等错误

谢谢大大提供的宝贝，以后有缘江湖见

DIYgod commented:
如文档里所说，puppeteer 性能较差且消耗较多资源，使用前请确保 HTML 和接口无法获取数据

貌似没看到从 HTML 和接口获取数据困难的情况

sunshinenny commented:
在使用puppeteer之前经过了长期的“勘探”

百度这个网页是动态生成的。看了好久没有找到可以调用的API。（在Network中发现能调用的API只有一个[API ](http://zhaopin.baidu.com/api/qzasync?query=司机&sort_type=1&detailmode=open&rn=20&city=上海&pn=40)）但是很神奇的是查询出来的对象和关键词几乎没有关系（；￣ェ￣）
而且也没办法通过got直接访问。所以我才使用的puppeteer…
我明天再尝试一下正常渠道获取信息。🤖️

DIYgod commented:
> 在使用puppeteer之前经过了长期的“勘探”
> 
> 百度这个网页是动态生成的。看了好久没有找到可以调用的API。（在Network中发现能调用的API只有一个[API ](http://zhaopin.baidu.com/api/qzasync?query=%E5%8F%B8%E6%9C%BA&sort_type=1&detailmode=open&rn=20&city=%E4%B8%8A%E6%B5%B7&pn=40)）但是很神奇的是查询出来的对象和关键词几乎没有关系（；￣ェ￣）
> 而且也没办法通过got直接访问。所以我才使用的puppeteer…
> 我明天再尝试一下正常渠道获取信息。🤖️

数据就在 HTML 里...

这个 pr 先关掉了

sunshinenny commented:
嗯………
但是我抓下来的就只有这样的源码…
这个暂时搁置了吧ᶘ ᵒᴥᵒᶅ 以后再做…
```html
<!DOCTYPE html>
<html>
<!--STATUS OK-->
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="description" content="百度帐号是登录所有百度系产品的通行证，登录后还可以在帐户管理页管理/修改您的个人信息，包括修改密码、绑定手机、身份认证等">
<title>
登录百度帐号
</title>
<link rel="shortcut icon" href="https://www.baidu.com/favicon.ico" type="image/x-icon" />
<link rel="icon" sizes="any" mask href="https://www.baidu.com/img/baidu.svg">                <script type="text/javascript">
            window.wpo={start:new Date*1,pid:109,page:'passport'};
            var _hmt = _hmt || [];
        </script>
<link href="/static/passpc-base/css/base.css?cdnversion=1560855391" type="text/css" rel="stylesheet" />        <link href="/static/passpc-base/css/ui/ui.css?cdnversion=1560855391" type="text/css" rel="stylesheet" />                        
<link href="/static/passpc-account/css/boot_login_7e48d99.css" type="text/css" rel="stylesheet" />
<!--[if lte IE 9]>
<style style="text/javascript">
.pass-form-item .pass-clearbtn:hover{
    background-position:-27px -95px;
    transform: rotate(0deg);
}
.pass-suggestion-list li.pass-item-suggsetion a:hover{
    background-position:-28px -115px; 
    transform: rotate(0deg);
}
</style>
<![endif]-->
<!--[if IE 8]>
<style style="text/javascript">
.pass-form-item-memberPass{
    margin-top: -5px;
}
</style>
<![endif]-->
</head>
<body>
<div id="wrapper" class="">
<div id="head">
                   <div class="mod-header">
<a href="http://www.baidu.com/"><img src="/static/passpc-base/img/logo/logo.gif" alt="logo"></a>
</div>
                </div>
<div id="nav">
                <div class="nav-2">
<div class="mod-nav clearfix">
<h1 class="page-type-notab"></h1>
</div>
</div>
        </div>
<div id="content">
<div class="clearfix login-container">
<div class="login-form">
<div class="login-wrapper">
<div id="login"></div>
</div>
</div>
</div>
<div class="app-download-link">
<a href="https://passport.baidu.com/export/app/index.html" target="_blank"><img src="https://passport.baidu.com/export/app/img/qrcode_android.png" width="66" height="66"></a>
</div><input type="hidden" value="https://zhaopin.baidu.com/quanzhi?city=%E5%8C%97%E4%BA%AC&query=%E5%8F%B8%E6%9C%BA&ua_from=pc" id="jsU" />
<input type="hidden" value="pp" id="jsProduct" />
<input type="hidden" value="" id="jsIdc" />
<input type="hidden" value="" id="jsSubpro" />
<input type="hidden" value="" id="jsNorealU" />
<input type="hidden" value="" id="jsOverseas" />
<input type="hidden" value="1" id="jsSms" />
<input type="hidden" value="" id="jsUserPwdLogin">
<input type="hidden" value="optional" id="jsAuthsiteAct" />
<input type="hidden" value="" id="jsAccountValue" />
<input type="hidden" value="廊坊" id="jsCurrentCity" />
</div>
<div id="foot">
                    <div class="mod-footer">
<div class="copy-box">2019&nbsp;&copy;Baidu</div>
</div>
                </div>
</div>
        <script type="text/javascript" src="/passApi/js/uni_wrapper.js?cdnversion=1560855391"></script>
<script type="text/javascript" src="/static/passpc-base/js/core.js?cdnversion=1560855391"></script>                            <script type="text/javascript" src="/static/passpc-base/js/ui/ui.js?cdnversion=1560855391"></script>
        <script type="text/javascript" src="https://passport.baidu.com/passApi/js/wrapper.js?v=1560855391"></script>
<script>function getVal(e){var t=document.getElementById(e).value;return t}var jsU=getVal("jsU"),jsProduct=getVal("jsProduct"),jsIdc=getVal("jsIdc"),jsSubpro=getVal("jsSubpro"),jsNorealU=getVal("jsNorealU"),jsAuthsiteAct=getVal("jsAuthsiteAct"),jsOverseas=getVal("jsOverseas"),jsSms=getVal("jsSms"),jsUserPwdLogin=getVal("jsUserPwdLogin"),_config={u:jsU,product:jsProduct,idc:jsIdc,subpro:jsSubpro,noreal_u:jsNorealU,staticpage:document.location.protocol+"//"+document.location.host+"/static/passpc-account/html/v3Jump.html",authsiteLogin:["qzone","tsina","weixin"],authsiteCfgLogin:{act:jsAuthsiteAct,display:"popup"},sms:jsSms,overseas:jsOverseas,qrcode:!0,qrcode_animation:1};document.charset&&-1==document.charset.toLowerCase().indexOf("utf")&&navigator.userAgent.indexOf("MSIE")>0&&(location.href=location.href+(location.href.indexOf("?")>-1?"&nocache=1":"?nocache=1"));var _accountValue=document.getElementById("jsAccountValue").value,current_city=document.getElementById("jsCurrentCity").value;</script>
<script type="text/javascript" src="/static/passpc-account/js/boot_login_1e30d1a.js"></script>                                        <script type="text/javascript">
        _hmt.push(['_setAutoPageview', false]);
        (function(){
            try{
                var tjstr = location.search,tjarr2 = [],tjresult = '';
                if(tjstr){
                    var tjarr = tjstr.slice(1).split('&');
                    for(var i = 0,len = tjarr.length;i<len;i++){
                        var query = tjarr[i].split('=')[0];
                        if(query != 't' && query != '_t'){
                            tjarr2.push(tjarr[i])
                        }
                    }
                }
                tjresult = location.pathname + (tjarr2.length > 0 ? ('?'+tjarr2.join('&')) : '')           
            }catch(e) {
                var tjresult = location.pathname
            }
            _hmt.push(['_trackPageview', tjresult]);
        })()
        var _bdhmProtocol = (("https:" == document.location.protocol) ? " https://" : " http://"); 
        document.write(unescape("%3Cscript src='" + _bdhmProtocol + "hm.baidu.com/h.js%3F90056b3f84f90da57dc0f40150f005d5' type='text/javascript'%3E%3C/script%3E"));
        </script>
<script type="text/javascript">wpo.tti=new Date*1;</script>
</body>
</html>
```
